### PR TITLE
Warn user when initial connection to Fedora fails and the URL does not e...

### PR DESCRIPTION
**Pull request by hectorcorrea**
Addresses issue https://github.com/projecthydra/active_fedora/issues/656

### CODE REVIEW

File: lib/active_fedora/fedora.rb [hunk 1]
```diff
@@ -26,6 +26,9 @@ def init_base_path
         ActiveFedora::Base.logger.info "Attempted to init base path `#{root_resource_path}`, but it already exists" if ActiveFedora::Base.logger
         return false
     rescue Ldp::NotFound
+      if !host.downcase.end_with?("/rest")
+        ActiveFedora::Base.logger.warn "Fedora URL (#{host}) does not end with /rest. This could be the problem. Check your fedora.yml config"
```
Comment (jcoyne) [inline lib/active_fedora/fedora.rb]: Switch "This could be the problem" to "This could be a problem"

**Comment by jcoyne** [comment]
This assumes they will never modify https://github.com/fcrepo4/fcrepo4/blob/3bfc11f2a3088f7435e47268faafb4f000ae50c0/fcrepo-webapp/src/main/webapp/WEB-INF/web.xml#L33 is that a safe assumption?

File: spec/unit/active_fedora_spec.rb [hunk 2]
```diff
@@ -15,6 +15,24 @@
...
   end
 
+  describe "validate Fedora URL" do 
+
+    let(:good_url) { ActiveFedora.fedora_config.credentials[:url] }
+    let(:bad_url) { good_url.gsub('/rest', '/') }
+
+    it "should connect OK" do 
+      expect(ActiveFedora::Base.logger).to_not receive(:warn)
+      ActiveFedora::Fedora.new(:url => good_url, :base_path => '/test')
```
Comment (mjgiarlo) [inline spec/unit/active_fedora_spec.rb]: Use Ruby 1.9-style hashes, pls, good sir. (Kill the hashrockets!)

File: spec/unit/active_fedora_spec.rb [hunk 3]
```diff
@@ -15,6 +15,24 @@
...
+    end
+
+    it "should not connect and warn" do 
+      expect {
+        expect(ActiveFedora::Base.logger).to receive(:warn)
```
Comment (mjgiarlo) [inline spec/unit/active_fedora_spec.rb]: Should this line be on the previous line, outside the `expect { ... }.to raise_error...`?

File: spec/unit/active_fedora_spec.rb [hunk 4]
```diff
@@ -15,6 +15,24 @@
...
+        ActiveFedora::Fedora.new(:url => bad_url, :base_path => '/test')
```
Comment (mjgiarlo) [inline spec/unit/active_fedora_spec.rb]: Use Ruby 1.9-style hashes, pls, good sir. (Kill the hashrockets!)

**Comment by jcoyne** [comment]
Another way to do this would be to get the resource at `host` and see if it is returning an RDF graph.

**Comment by mjgiarlo** [comment]
@jcoyne I'm not too concerned about the web.xml thing, because the warning only appears when an LDP error is raised, and the message is merely a suggestion about what might be an easy mistake to make for folks new to F4.

Though I wonder if another quick check might be taking a glance at the base_path value and making sure it exists?  That's tripped one or two of us up already.

**Comment by jcoyne** [comment]
@mjgiarlo this ticket addresses #656, you are bringing up issue #680

File: spec/unit/active_fedora_spec.rb (hunk 2 above)
Comment (hectorcorrea) [inline spec/unit/active_fedora_spec.rb]: @mjgiarlo They are now gone [insert sad :rocket: here]

File: spec/unit/active_fedora_spec.rb (hunk 4 above)
Comment (hectorcorrea) [inline spec/unit/active_fedora_spec.rb]: @mjgiarlo They are now gone [insert sad :rocket: here]

File: spec/unit/active_fedora_spec.rb (hunk 3 above)
Comment (hectorcorrea) [inline spec/unit/active_fedora_spec.rb]: @mjgiarlo It worked both ways, but I moved it out as you suggested to prevent confusion.

**Comment by hectorcorrea** [comment]
@mjgiarlo @jcoyne I am not to worried about users tweaking the web.xml because as Mike pointed out the warning is only raised if the connection fails. I suspect users that intentionally update the Fedora settings will know what to make out of the warning.

**Comment by mjgiarlo** [comment]
w00t, :rocket: is :bomb:

**Comment by flyingzumwalt** [comment]
:+1:  :wink:

**Pull request merged**